File: app/src/main/java/com/aegamesi/steamtrade/steam/SteamAPI.java
package com.aegamesi.steamtrade.steam;

import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class SteamAPI {

    private static Retrofit retrofit;
    private static final String BASE_URL = "https://store.steampowered.com/api/";

    public static Retrofit getApiInstance() {
        if (retrofit == null) {

            retrofit = new Retrofit.Builder()
                    .baseUrl(BASE_URL)
                    .addConverterFactory(GsonConverterFactory.create())
                    .build();
        }
        return retrofit;
    }
}

File: app/src/main/java/com/aegamesi/steamtrade/steam/SteamAPI.kt
package com.aegamesi.steamtrade.steam

import retrofit2.Retrofit
import retrofit2.converter.gson.GsonConverterFactory

object SteamAPI {

    private var retrofit: Retrofit? = null
    private const val BASE_URL = "https://store.steampowered.com/api/"

    val apiInstance: Retrofit
        get() {
            if (retrofit == null) {

                retrofit = Retrofit.Builder()
                        .baseUrl(BASE_URL)
                        .addConverterFactory(GsonConverterFactory.create())
                        .build()
            }
            return retrofit!!
        }
}

